File: benchmark/corpus/java/cases/vuln_format_string.java
import javax.servlet.http.HttpServletRequest;

public class VulnFormatString {
    String greet(HttpServletRequest request) {
        String username = request.getParameter("username");
        // ZS-JAVA-024: the FORMAT STRING itself is attacker-controlled, so the
        // user chooses the format specifiers. Note this is deliberately not
        // String.format("User: %s", username) — that idiom is safe and the
        // rule must not fire on it (see clean.java).
        return String.format("User: " + username);
    }
}

File: benchmark/corpus/csharp/cases/vuln_format_string.cs
public class VulnFormatString
{
    public string Greet()
    {
        var username = Request.QueryString["username"];
        // ZS-CS-019: the FORMAT STRING itself is attacker-controlled. The safe
        // string.Format("User: {0}", username) idiom must not fire — see clean.cs.
        return string.Format("User: " + username);
    }
}
